packages/remix/src/ssr/authenticateRequest.ts

```diff
@@ -41,7 +40,17 @@ export function authenticateRequest(args: LoaderFunctionArgs, opts: RootAuthLoad
     (opts as RootAuthLoaderOptionsWithExperimental).domain ||
     '';
 
-  const proxyUrl = getEnvVariable('CLERK_PROXY_URL') || (context?.CLERK_PROXY_URL as string) || opts.proxyUrl || '';
+  const isSatellite =
+    getEnvVariable('CLERK_IS_SATELLITE') === 'true' ||
+    (context?.CLERK_IS_SATELLITE as string) === 'true' ||
+    handleIsSatelliteBooleanOrFn((opts as RootAuthLoaderOptionsWithExperimental).isSatellite, new URL(request.url)) ||
```

Comment:
:upside_down_face: Isn't it clearer if we simply do something like 
```
    getEnvVariable('CLERK_IS_SATELLITE') === 'true' ||
    (context?.CLERK_IS_SATELLITE as string) === 'true' ||
    opts.isSatellite === true || 
    opts.isSatellite(url);
```
If `isSatellite` is a function it will never be strictly equal to `true`.

Reply:
Unfortunately this will not work and we will still need to do explicit type casting or check if `isSatellite ` is type of function because at this point it could either be `false` or `undefined`. And now this gets messy.

That was the reason why I moved this logic to a function

Reply:
You're right. The code sample I wrote was not meant to be used as is, just to illustrate my point. :smile: 

Feel free to leave things as they are!

Reply:
This looks fine to me. 
Please ignore this comment as this is extremely nitpicky, I'm just adding it for the sake of discussion. The only change I'd make would be to change this to 
```
    isSatellite(req, opts)
```
and move the implementation details inside the function because I totally agree with what you said :) :

> That was the reason why I moved this logic to a function

Reply:
>  isSatellite(req, opts)
That is very similar to what i had before in terms of a parameters. Naming was deferent.

Would it make sense to encapsulate the whole logic of 
``` typescript
getEnvVariable('CLERK_IS_SATELLITE') === 'true' ||
    (context?.CLERK_IS_SATELLITE as string) === 'true' ||
    handleIsSatelliteBooleanOrFn((opts as RootAuthLoaderOptionsWithExperimental).isSatellite, new URL(request.url)) ||
    false;
```    
inside a `isSatellite(args: LoaderFunctionArgs, opts)` helper function ?